Description: Fix out of source build to cope with Debian methods
Author: Fathi Boudra <fabo@debian.org>
Last-Update: 2008-05-01

--- a/kcm_gtk/CMakeLists.txt
+++ b/kcm_gtk/CMakeLists.txt
@@ -16,7 +16,7 @@ KDE4_ADD_UI_FILES(kcm_gtk4_SOURCES ${kcm
 
 KDE4_ADD_PLUGIN(kcm_gtk4 ${kcm_gtk4_SOURCES})
 
-INCLUDE_DIRECTORIES(${KDE4_INCLUDE_DIR})
+INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR} ${KDE4_INCLUDE_DIR})
 LINK_DIRECTORIES(${KDE4_LIB_DIR})
 
 TARGET_LINK_LIBRARIES(kcm_gtk4 ${KDE4_KDECORE_LIBRARY} ${KDE4_KDEUI_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY})
--- a/po/CMakeLists.txt
+++ b/po/CMakeLists.txt
@@ -1,46 +1,23 @@
-# .po to .gmo stuff
+FIND_PROGRAM(GETTEXT_MSGFMT_EXECUTABLE msgfmt)
+
 FILE(GLOB _pofiles *.po)
 
-FOREACH(_file ${_pofiles})
-	GET_FILENAME_COMPONENT(_file_we ${_file} NAME_WE)
-	SET(_out "${_file_we}.gmo")
-	SET(_in "${_file_we}.po")
-	
-	ADD_CUSTOM_COMMAND(
-		OUTPUT ${_out}
-		COMMAND ${MSGFMT_EXECUTABLE} -o ${_out} ${_in}
-		DEPENDS ${_in}
-	)
-	
-	INSTALL(
-		FILES ${_out}
-		DESTINATION share/locale/${_file_we}/LC_MESSAGES/
-		RENAME gtkqtengine.mo
-	)
-	
-	SET(_outputs ${_outputs} ${_out})
-ENDFOREACH(_file)
-
-ADD_CUSTOM_TARGET(
-	pofiles ALL
-	DEPENDS ${_outputs}
-)
-
-
-# Stuff to generate the .pot
-FILE(GLOB POT_UIFILES ../kcm_gtk/*.ui)
-SET(POT_SOURCES ../kcm_gtk/kcmgtk.cpp ../kcm_gtk/firefoxfix.cpp ../kcm_gtk/gtkrcfile.cpp ../kcm_gtk/searchpaths.cpp)
-SET(POT_OUTPUT gtkqtengine.pot)
-
-ADD_CUSTOM_COMMAND(
-	OUTPUT _${POT_OUTPUT}
-	COMMAND ${EXTRACTRC_PATH} ${POT_UIFILES} > rc.cpp
-	COMMAND ${XGETTEXT_PATH} --foreign-user -C -ci18n -ki18n -ktr2i18n
-		-kI18N_NOOP -kI18N_NOOP2 -kaliasLocale -o ${POT_OUTPUT} rc.cpp ${POT_SOURCES}
-	COMMAND rm rc.cpp
-)
-
-ADD_CUSTOM_TARGET(
-	extract_messages
-	DEPENDS _${POT_OUTPUT}
-)
+FOREACH(_current_PO_FILE ${_pofiles})
+   SET(_addToAll ALL)
+   SET(_installDest ${LOCALE_INSTALL_DIR})
+
+   GET_FILENAME_COMPONENT(_basename ${_current_PO_FILE} NAME_WE)
+      SET(_gmoFile ${CMAKE_CURRENT_BINARY_DIR}/${_basename}.gmo)
+   message(STATUS gmo file: ${_gmoFile})
+      add_custom_command(OUTPUT ${_gmoFile}
+            COMMAND ${GETTEXT_MSGFMT_EXECUTABLE} -o ${_gmoFile} ${_current_PO_FILE}
+            WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+            DEPENDS ${_current_PO_FILE}
+         )
+
+   INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/${_basename}.gmo DESTINATION ${_installDest}/${_basename}/LC_MESSAGES/ RENAME kcmgtk4.mo)
+
+   LIST(APPEND _gmoFiles ${_gmoFile})
+ENDFOREACH(_current_PO_FILE)
+
+ADD_CUSTOM_TARGET(pofiles ${_addToAll} DEPENDS ${_gmoFiles})
